### PR TITLE
feat: add Int16/Int32/Int64

### PR DESCRIPTION
**Pull request by hargoniX**
This adds all fixed width integers with the exception of `ssize_t` so the code is quick to review as everything just behaves the same.

### CODE REVIEW

File: src/include/lean/lean.h [hunk 1]
```diff
@@ -1981,6 +1993,429 @@ static inline uint8_t lean_int8_dec_le(uint8_t a1, uint8_t a2) {
...
+}
+
+static inline uint16_t lean_int16_neg(uint16_t a) {
+    int16_t arg = (int16_t)a;
+
+    return (uint16_t)(-arg);
+}
+
+static inline uint16_t lean_int16_add(uint16_t a1, uint16_t a2) {
+    int16_t lhs = (int16_t) a1;
```
Comment (TwoFX) [inline src/include/lean/lean.h:2040]: You have some inconsistency about whether you have a space or not in the cast, could you clean that up?

File: src/include/lean/lean.h [hunk 2]
```diff
@@ -1981,6 +1993,429 @@ static inline uint8_t lean_int8_dec_le(uint8_t a1, uint8_t a2) {
...
+    int16_t rhs = (int16_t) a2;
+
+    return (uint16_t)(rhs == 0 ? 0 : lhs / rhs);
+}
+
+static inline uint16_t lean_int16_mod(uint16_t a1, uint16_t a2) {
+    int16_t lhs = (int16_t) a1;
+    int16_t rhs = (int16_t) a2;
+
+    return (uint16_t)(rhs == 0 ? 0 : lhs % rhs);
```
Comment (TwoFX) [inline src/include/lean/lean.h:2071]: This is a miscompilation (also present for `Int8`).

File: src/Init/Data/SInt/Basic.lean [hunk 3]
```diff
@@ -11,7 +11,6 @@ This module contains the definition of signed fixed width integer types as well
 and bitwise operations on top of it.
 -/
 
-
```
Comment (tobiasgrosser) [inline src/Init/Data/SInt/Basic.lean:14]: This seems unrelated.

File: src/include/lean/lean.h [hunk 4]
```diff
@@ -1961,26 +1973,449 @@ static inline uint8_t lean_int8_complement(uint8_t a) {
...
+    int16_t rhs = (int16_t)a2;
+
+    return (uint16_t)(lhs ^ rhs);
+}
+
+static inline uint16_t lean_int16_shift_right(uint16_t a1, uint16_t a2) {
+    int16_t lhs = (int16_t)a1;
+    int16_t rhs = (int16_t)a2;
+
+    return (uint16_t)(lhs >> (rhs % 16));
```
Comment (TwoFX) [inline src/include/lean/lean.h:2099]: This is UB and I am observing different results locally and on live.lean-lang.org, so we need to avoid shifting by negative amounts.

**Review by TwoFX — APPROVED** [review]
LGTM except for the request for documentation.
